database/seeders/TopicSeeder.php

```diff
@@ -4,29 +4,63 @@
 
 use App\Models\Topic;
 use Illuminate\Database\Seeder;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
 
-class TopicSeeder extends Seeder
-{
+class TopicSeeder extends Seeder {
     /**
      * Run the database seeds.
      *
      * @return void
      */
-    public function run()
-    {
-        $topics = [
-            __('Accessible consultation'),
-            __('Intersectional outreach'),
-            __('Contracts'),
-            __('Privacy'),
-            __('Disability knowledge'),
-        ];
-
-        foreach ($topics as $topic) {
-            Topic::firstOrCreate([
-                'name->en' => $topic,
-                'name->fr' => trans($topic, [], 'fr'),
-            ]);
+    public function run() {
+
+        // option to use environment to restore or backup to different environment files
+        if (config('seeder.environment') !== null && in_array(config('seeder.environment'), ['production', 'staging', 'local']) === true) {
+            $environment = config('seeder.environment');
+        } else {
+            $environment = config('app.env');
+        }
+
+        // check if there is a JSON file that has stored data for the seeder
+        if (in_array(config('app.env'), ['testing', 'production']) !== true) {
+
+            // if trucate was set via seeder restore command then truncate the table prior to seeding data
+            if (config('seeder.truncate')) {
+                DB::statement("SET foreign_key_checks=0");
+                Topic::truncate();
+                DB::statement("SET foreign_key_checks=1");
+            }
+
+            if (Storage::disk('seeds')->exists(sprintf("topics.%s.json", $environment))) {
+
+                $topics = json_decode(Storage::disk('seeds')->get(sprintf("topics.%s.json", $environment)), true);
+
+                foreach ($topics as $topic) {
+                    Topic::firstOrCreate([
+                        'name' => json_decode($topic['name'], true),
+                    ]);
+                }
+            } else {
+                print("Seeder file wasn't found, using default values\r\n");
+                $topics = [
+                    __('Accessible consultation'),
+                    __('Intersectional outreach'),
+                    __('Contracts'),
+                    __('Privacy'),
+                    __('Disability knowledge'),
+                ];
+
+                foreach ($topics as $topic) {
+                    Topic::firstOrCreate([
+                        'name->en' => $topic,
+                        'name->fr' => trans($topic, [], 'fr'),
+                    ]);
+                }
+            }
+        } else {
+            $environment = config('app.env');
+            printf("Seeder cannot be run on environment: %s\r\n", $environment);
```

Comment:
If we still need this can we change it to something like the following so we know which seeder didn't run.

```suggestion
            printf("%s cannot be run on environment: %s\r\n", self::class, $environment);
```

The same would apply for all other cases.

Reply:
We are removing the check altogether.